packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -783,6 +750,50 @@ class CupertinoTextField extends StatefulWidget {
   /// {@macro flutter.services.TextInputConfiguration.enableIMEPersonalizedLearning}
   final bool enableIMEPersonalizedLearning;
 
+  /// {@macro flutter.widgets.EditableText.contextMenuBuilder}
+  ///
+  /// If not provided, will build a default menu based on the platform.
+  ///
+  /// See also:
+  ///
+  ///  * [DefaultCupertinoTextSelectionToolbar], which is built by default.
+  final EditableTextToolbarBuilder? contextMenuBuilder;
+
+  /// The platform that the current `defaultTargetPlatform` will be treated as.
+  ///
+  /// The Cupertino library has no access to Material, so it cannot build things
+  /// like the Material text selection toolbars on the platforms that use them.
+  /// Instead it will treat those non-Cupertino platforms as if they are the
+  /// platform returned here.
+  ///
+  /// For example, using a CupertionTextField on an Android device will show the
+  /// iOS text selection toolbar.
```

Comment:
Should this be a correct behavior? I would imagine people will still expect to see Android text selection toolbar.

Reply:
I agree, but it's my understanding that we have to do this when we're totally within the Cupertino library and don't have access to Material.  This is the pattern that I've followed for other stuff like this in the past.

It is possible for users to pass in an adaptive context menu to their own CupertinoTextField if they are using Material too, though.

Reply:
I agree, that the user would expect to see the Android toolbar on an Android device regardless of the TextField used, maybe there could be an example on how to pass a Material text selection toolbar to a CupertinoTextField for these cases?